### PR TITLE
Add constant rate controller with patience

### LINKED ISSUE #13: Add constant learning rate controller with patience
Patience is how long the rate controller will continue returning new learning rate values based on the trajectory of the losses.  If the losses are not continuing to go down after a while you want to end training.  If you continue training you might overfit the model leading to worse generalization than if you had stopped training earlier.

### PR DESCRIPTION
**Pull request by davidlmorton**
Closes #13

### CODE REVIEW

**Comment by davidlmorton** [comment]
retest this please

**Comment by davidlmorton** [comment]
retest this please

**Comment by davidlmorton** [comment]
retest this please

**Comment by davidlmorton** [comment]
retest this please

**Comment by davidlmorton** [comment]
retest this please

**Comment by davidlmorton** [comment]
retest this please
